stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt

```diff
@@ -210,6 +210,29 @@ public class MessageInputViewModel @JvmOverloads constructor(
         sendMessageInternal(message.apply(messageTransformer))
     }
 
+    /**
+     * Resends the message that failed moderation. If nothing is passed for [newMessageText] that means that the
+     * message wasn't edited and the original will be resent.
+     *
+     * @param oldMessage The [Message] which failed moderation.
+     * @param newMessageText Optional new text for the message. If null old message is resent.
+     */
+    public fun resendModeratedMessage(
+        oldMessage: Message,
+        newMessageText: String? = null
+    ) {
+        chatClient.deleteMessage(oldMessage.id, true).enqueue()
+        if (oldMessage.attachments.isEmpty()) {
+            sendMessage(newMessageText ?: oldMessage.text) {
+                parentId = oldMessage.parentId
+            }
+        } else {
+            sendMessageWithCustomAttachments(newMessageText ?: oldMessage.text, oldMessage.attachments) {
```

Comment:
Is this covering only custom attachments or any? Does this name make sense for default attachments too?

Reply:
Since the message that was sent already has attachments parsed, this function does not do any logic upon the attachments. Calling `sendMessageWithAttachments` would not work properly. We can add another function that relies upon this one, but shouldn't change the name of this one.